cli/azd/pkg/input/progress_log.go

```diff
@@ -118,27 +118,29 @@ func (p *progressLog) Start() {
 
 // Stop clears the screen from any previous output and clear the buffer.
 // Calling Stop() before Start() is a no-op.
-func (p *progressLog) Stop() {
+func (p *progressLog) Stop(keepLogs bool) {
 	if p.output == nil {
 		return
 	}
 
 	p.outputMutex.Lock()
 	defer p.outputMutex.Unlock()
 
-	p.clearContentAndFlush()
-
-	// title
-	tm.MoveCursorUp(2)
-	clearLine()
+	if !keepLogs {
+		p.clearContentAndFlush()
 
-	// header
-	if p.header != "" {
+		// title
 		tm.MoveCursorUp(2)
 		clearLine()
-	}
 
-	tm.Flush()
+		// header
+		if p.header != "" {
+			tm.MoveCursorUp(2)
+			clearLine()
+		}
+
+		tm.Flush()
+	}
 	p.output = nil
```

Comment:
A really naive question here... Do we at least need to do something along the lines of `Flush` here to make output show up before clearing it?

Reply:
The flush() call happens for every Write (at the end), to refresh the output after updating the buffer.
When the component is stopped, we only need to call flush() if we make changes to the current buffer and need to re-paint the screen. For the case of keeping the logs, we want to skip this step